src/main/java/org/thingsboard/gateway/extensions/sigfox/conf/mapping/SigfoxDeviceMapping.java

```diff
@@ -0,0 +1,110 @@
+package org.thingsboard.gateway.extensions.sigfox.conf.mapping;
+
+import com.jayway.jsonpath.DocumentContext;
+import com.jayway.jsonpath.JsonPath;
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.util.StringUtils;
+import org.thingsboard.gateway.extensions.opc.conf.mapping.DeviceMapping;
+import org.thingsboard.gateway.service.data.DeviceData;
+import org.thingsboard.server.common.data.kv.*;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+@Data
+@Slf4j
+public class SigfoxDeviceMapping {
+
+    public static final Pattern TAG_PATTERN = Pattern.compile("\\$\\{(.*?)\\}");
+
+    private String deviceNameJsonExpression;
+    private final List<AttributesMapping> attributes;
+    private final List<TimeseriesMapping> timeseries;
+
+    public DeviceData parseBody(String body) {
+        try {
+            DocumentContext document = JsonPath.parse(body);
+            long ts = System.currentTimeMillis();
+            String deviceName = eval(document, deviceNameJsonExpression);
+            if (!StringUtils.isEmpty(deviceName)) {
+                List<KvEntry> attrData = getKvEntries(JsonPath.parse(body), attributes);
+                List<TsKvEntry> tsData = getKvEntries(JsonPath.parse(body), timeseries).stream()
+                        .map(kv -> new BasicTsKvEntry(ts, kv))
+                        .collect(Collectors.toList());
+                return new DeviceData(deviceName, attrData, tsData);
+            }
+        } catch (Exception e) {
+            log.error("Exception occurred while parsing json request body [{}]", body, e);
+            throw e;
+        }
+        return null;
+    }
+
+    private List<KvEntry> getKvEntries(DocumentContext document, List<? extends SigfoxKVMapping> mappings) {
+        List<KvEntry> result = new ArrayList<>();
+        if (mappings != null) {
+            for (SigfoxKVMapping mapping : mappings) {
+                String key = eval(document, mapping.getKey());
+                String strVal = eval(document, mapping.getValue());
+                switch (mapping.getType().getDataType()) {
+                    case STRING:
+                        result.add(new StringDataEntry(key,
+                                applyTransformer((DataValueTransformer<String>) mapping.getTransformer(), strVal)));
```

Comment:
no need to cast if you will add those four methods for each type to transformer interface